src/views/Generator/RuleEditor/HeaderSelect.hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { Filter } from '@/types/rules'
+import { matchFilter } from '@/rules/utils'
+import { ProxyData } from '@/types'
+import { useMemo } from 'react'
+
+export function useHeaderOptions(
+  recording: ProxyData[],
+  extractFrom: 'request' | 'response',
+  filter: Filter
+) {
+  return useMemo(() => {
+    const filteredRequests = recording.filter((entry) =>
+      matchFilter(entry.request, filter)
+    )
+
+    const headers = filteredRequests.flatMap(
+      (request) => request?.[extractFrom]?.headers ?? []
+    )
```

Comment:
I noticed some headers are listed in the dropdown but not generated in the script so selecting them doesn't have any effect. We could exclude them from the dropdown as well to keep consistency.

https://github.com/grafana/k6-studio/blob/5faa0405183b9b1a636ba71c652db65affd4c20e/src/codegen/codegen.ts#L189-L191

Reply:
Good call! Updated 👌